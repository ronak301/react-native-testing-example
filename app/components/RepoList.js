import React, { Component } from 'react';
import {
  ListView,
  StyleSheet,
  Text,
} from 'react-native';

import RepoItem from './RepoItem';

const ds = new ListView.DataSource({rowHasChanged: (r1, r2) => r1.id !== r2.id});

const RepoList = (props) => {
  const { repos, selected, selectRepo } = props;
  const dataSource = ds.cloneWithRows(repos);
  return (
    <ListView
      dataSource={dataSource}
      renderRow={(repo) => (
        <RepoItem
          isSelected={repo.id === selected}
          repo={repo}
          selectRepo={selectRepo}
        />
      )}
      contentContainerStyle={styles.list}
    />
  );
};

RepoList.propTypes = {
  repos: React.PropTypes.array.isRequired,
  selected: React.PropTypes.number,
  selectRepo: React.PropTypes.func,
};

export const styles = StyleSheet.create({
  list: {
    paddingVertical: 40,
  }
});

export default RepoList;