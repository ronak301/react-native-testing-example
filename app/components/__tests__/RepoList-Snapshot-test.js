import React from 'react';
import renderer from 'react-test-renderer';   // renders React-Native components in form of pure javascript objects.

import RepoList from '../RepoList';
import { repos } from '../../../config/jest/mockData';

/*
for testing a List View Component we give it MOCK_DATA as props,
which replicates the original data. The snapshot would contain the
breakdown of the whole ListView Component in terms of functions and
Values that we have given in MOCK_DATA, it would also show the styles
applied to different react components. All these will be saved in a
.snap file. Everytime a test fails, we can check the diff in .snap file
and accordingly select whether the change was desired or not and accordingly
modify the snapshot for future tests.
*/

/*
Here, since selectRepo prop requires a function, we mock it by using jest's
mock function prototype.
*/

it('renders a RepoList using Snapshots', () => {
  const component = renderer.create(
    <RepoList
      repos={repos}
      selectRepo={jest.fn()}
    />
  );
  expect(component).toMatchSnapshot();
});
