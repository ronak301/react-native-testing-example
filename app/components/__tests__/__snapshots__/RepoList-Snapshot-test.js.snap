exports[`test renders a RepoList using Snapshots 1`] = `
<ScrollView
  contentContainerStyle={
    Object {
      "paddingVertical": 40
    }
  }
  dataSource={
    ListViewDataSource {
      "items": 3
    }
  }
  initialListSize={10}
  onContentSizeChange={[Function bound _onContentSizeChange]}
  onEndReachedThreshold={1000}
  onKeyboardDidHide={undefined}
  onKeyboardDidShow={undefined}
  onKeyboardWillHide={undefined}
  onKeyboardWillShow={undefined}
  onLayout={[Function bound _onLayout]}
  onScroll={[Function bound _onScroll]}
  pageSize={1}
  removeClippedSubviews={true}
  renderRow={[Function renderRow]}
  scrollEventThrottle={50}
  scrollRenderAheadDistance={1000}
  stickyHeaderIndices={Array []}>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function bound touchableHandleResponderGrant]}
    onResponderMove={[Function bound touchableHandleResponderMove]}
    onResponderRelease={[Function bound touchableHandleResponderRelease]}
    onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
    onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
    onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent"
        },
        undefined
      ]
    }
    testID={undefined}>
    <View
      style={
        Object {
          "paddingHorizontal": 24,
          "paddingVertical": 8
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
            "paddingBottom": 4
          }
        }>
        Test 1
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "paddingBottom": 8
          }
        }>
        50000 stars
      </Text>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function bound touchableHandleResponderGrant]}
    onResponderMove={[Function bound touchableHandleResponderMove]}
    onResponderRelease={[Function bound touchableHandleResponderRelease]}
    onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
    onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
    onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent"
        },
        undefined
      ]
    }
    testID={undefined}>
    <View
      style={
        Object {
          "paddingHorizontal": 24,
          "paddingVertical": 8
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
            "paddingBottom": 4
          }
        }>
        Test 2
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "paddingBottom": 8
          }
        }>
        10000 stars
      </Text>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function bound touchableHandleResponderGrant]}
    onResponderMove={[Function bound touchableHandleResponderMove]}
    onResponderRelease={[Function bound touchableHandleResponderRelease]}
    onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
    onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
    onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent"
        },
        undefined
      ]
    }
    testID={undefined}>
    <View
      style={
        Object {
          "paddingHorizontal": 24,
          "paddingVertical": 8
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
            "paddingBottom": 4
          }
        }>
        Test 3
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "paddingBottom": 8
          }
        }>
        499 stars
      </Text>
    </View>
  </View>
</ScrollView>
`;
